modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,829 @@
+	description = "A portable device which manually overrides and controls a cyborg's OS."
+	prereq_ids = list("robotics", "sec_basic")
+	design_ids = list(
+		"cyborg_control_bay",
+	)
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 7500)
+	discount_experiments = list(/datum/experiment/ordnance/explosive/highyieldbomb = 5000)
+
+/datum/design/cyborg_control_bay
+	name = "Deployable Control-Bay"
+	id = "cyborg_control_bay"
+	materials = list(/datum/material/iron = 1000, /datum/material/glass = 1000, /datum/material/gold = 2000, /datum/material/silver = 2000)
+	build_path = /obj/item/grenade/cyborg_control_bay_undeployed
+	build_type = AUTOLATHE | PROTOLATHE
+	departmental_flags = DEPARTMENT_BITFLAG_SECURITY | DEPARTMENT_BITFLAG_SCIENCE
+	category = list(RND_CATEGORY_EQUIPMENT)
+
+//	The control bay in its deployed state
+/obj/structure/bed/cyborg_control_bay
+	name = "cyborg control-bay"
+	desc = "A nimble device with an electric port suited to connect with station-approved cyborgs. Once attached the control-bay supersedes the cyborg's OS hierarchy to allow for maintenance."
+	icon = 'modular_skyrat/modules/deployables/icons/deployable.dmi'
+	icon_state = "up"
+	anchored = FALSE
+	resistance_flags = FIRE_PROOF | FREEZE_PROOF
+	flags_1 = NODECONSTRUCT_1
+
+	bolts = FALSE
+	///	The cyborg currently buckled to the cyborg_control_bay
+	var/mob/living/silicon/robot/buckled_cyborg
+	///	If the cyborg_control_bay is deployed or not
+	var/deployed = TRUE
+	///	Wether or not the machine is locking the cyborg
+	var/locked = FALSE
+	///	To distinguish if the machine is pumping or draining
+	var/enabled_function = NONE
+	///	Amount of power drained from the cyborg, which we are now storing
+	var/power_storage = 0
+
+/obj/structure/bed/cyborg_control_bay/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CLICK_ALT, .proc/check_alt_clicked_radial)
+	update_appearance(UPDATE_ICON)
+
+/obj/structure/bed/cyborg_control_bay/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_CLICK_ALT)
+	if(buckled_cyborg)
+		unlock()
+		buckled_cyborg = null
+
+//	A destruction call to explode if the bay held a decent amount of power
+/obj/structure/bed/cyborg_control_bay/atom_destruction(damage_flag)
+	var/turf/debris = get_turf(src)
+	if(power_storage >= EXPL_MIN_REQ)
+		explosion(src, -1, (power_storage / EXPL_CHARGE_MOD), (power_storage / EXPL_CHARGE_MOD), (power_storage / EXPL_CHARGE_MOD), -1)
+
+	new /obj/effect/decal/cleanable/robot_debris(debris)
+	return ..()
+
+/obj/structure/bed/cyborg_control_bay/update_overlays()
+	. = ..()
+	. += emissive_appearance(icon, "[icon_state]-em", alpha = src.alpha)
+
+/obj/structure/bed/cyborg_control_bay/examine(mob/user)
+	. = ..()
+	if(buckled_cyborg)
+		. += span_notice("Alt+Click to operate.")
+	else
+		. += span_notice("Alt+Click to undeploy.")
+
+	if(power_storage)
+		. += span_notice("Its power storage gauge indicates that it is currently storing <b>[power_storage] J</b> out of a maximum of <b>[MAX_POWER] J</b>.")
```

Comment:
The `display_joules(amount)` proc will convert it to the appropriate metric unit and add the unit suffix (eg to kJ or MJ)